Add nickname

tayyaba17 opened the pull request:
I added the nickname capability but when the user clicks the submit button, it redirects them to "/home" and I can't figure out where to change that... Let me know if you see anything that may help

tayyaba17 committed:
Add email and nickname parameters.

tayyaba17 committed:
Add email and nickname parameters.

tayyaba17 committed:
Add nickname capability in doGet() with helper function and doPost() method.

tayyaba17 committed:
Add create nickname button.

tayyaba17 committed:
Add function setNicknameForm() to open login page to set nickname.

peizhaoou commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -108,7 +113,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       numberOfCommentsToDisplay = getNumberOfCommentsToDisplay(request);
 
-      // Get the URL of the image that the user uploaded to Blobstore.
+      // Get the URL of  the image that the user uploaded to Blobstore.
```
Nit: redundant space?

peizhaoou commented on portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java:
```diff
@@ -18,24 +25,68 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
+      return;
+    }
+
+    String nickname = request.getParameter("nickname");
```
For string constants like "nickname", it would be good to make it a constant for better reuse, e.g.:

private final String NICKNAME = "nickname";

and then use NICKNAME here.

peizhaoou commented on portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java:
```diff
@@ -18,24 +25,68 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
+  /**
+   * Returns the nickname of the user with id, or empty String if the user has not set a nickname.
+   */
+  private String getUserNickname(String id) {
```
Is the function argument "id" never used?

martiKallas commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -108,7 +113,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       //Get number of comments.
```
nit: you probably don't need so many comments. Comments are good, but they are normally used for describing APIs/functions, describing code that isn't clear, or explaining choices that aren't clear. 

This comment (and some others in this file) isn't really needed because the code is easy to understand by reading the variable and function names. This is called 'self-documenting' code.

tayyaba17 replied:
I figured the redirect problem out earlier! It was redirecting it to an old file that I decided to remove. Will remove extra comments, thanks!

martiKallas reviewed (commented):
For the problem with redirecting to the home page, try thinking through the execution path. I'm not sure if this is the actual problem, but I think it might be. Where does the user get directed when they hit the submit nickname button? Where does that code direct them?

tayyaba17 committed:
Remove unnecessary comments.

tayyaba17 committed:
Update entity inside  post method to properly keep track of nickname. Update string constant naming. Update functuon getUserNickname to set nickname parameter by using id/
.

tayyaba17 commented:
I just updated some of the files based on the comments. I also fixed the redirect error. I think the authentication portion is set for now. When we work on displaying blogposts we should display the nickname of the user.

peizhaoou commented on portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java:
```diff
@@ -34,8 +34,12 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
       String nickname = getUserNickname(userService.getCurrentUser().getUserId());
+    
+      if (nickname == ""){ out.println("<p>You don't have a nickname</p>");
```
A couple comments:

(1) for nickname == "", are you supposed to check emptiness? If so, I think you should use nickname.isEmpty(). This right now is comparing reference equality.

(2) Use a new line after {, and add a space between ){.

peizhaoou commented on portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java:
```diff
@@ -34,8 +34,12 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       String nickname = getUserNickname(userService.getCurrentUser().getUserId());
+    
+      if (nickname == ""){ out.println("<p>You don't have a nickname</p>");
+      } else { out.println("<p>Your current nickname is " + nickname + ". If you'd like to change it, enter a new one below.</p>");}   
```
Same here: start a new line.

peizhaoou commented on portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java:
```diff
@@ -60,33 +64,31 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     }
 
-    String nickname = request.getParameter("nickname");
+    String NICKNAME = request.getParameter("nickname");
```
Sorry I was probably confusing for using constants. I meant having NICKNAME as a class member and use that here, e.g.:

private final static String NICKNAME = "nickname";

In the function body, use it like request.getParameter(NICKNAME)

tayyaba17 replied:
so after I have the class member NICKNAME, would the variables inside the functions be named "nickname" ? What's the best way to do that?

peizhaoou replied:
So per the Java style guide, constant variables would be named with everything capitalized. It's just a variable, but it's like a class-level global variable that you can refer to in your code. The motivation behind this is that if someday you need to change the parameter name from "nickname" to "alias", you can just change one place.

peizhaoou commented on portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java:
```diff
@@ -60,33 +64,31 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     }
-    String nickname = (String) entity.getProperty("nickname");
-    return nickname;
+    String NICKNAME = (String) entity.getProperty("nickname");
```
Same here.

tayyaba17 committed:
Created NICKNAME as class member.

peizhaoou reviewed (approved):
Thanks Tayyaba!

tayyaba17 committed:
Fix nit details.

tayyaba17 committed:
Update post button text.
